### PR TITLE
Set up Gradle for publishing

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Tested with `./gradlew publishToMavenLocal`

### CODE REVIEW

**Comment by artem-zinnatullin** [comment]
Would appreciate your review @vanniktech

File: build.gradle [hunk 1]
```diff
@@ -11,15 +11,46 @@ buildscript {
...
 
+group = 'com.lyft.domic'
+version = projectVersion()
+
 allprojects {
     repositories {
-        jcenter()
         google()
+        jcenter()
     }
```
Comment (rcd27) [inline build.gradle]: Artem, why would you change the places of repositories? I heard something about Kotlin wants them to be in that order 8)

File: build.gradle [hunk 2]
```diff
@@ -11,15 +11,46 @@ buildscript {
...
 }
 
+subprojects {
```
Comment (vanniktech) [inline build.gradle]: Can't you do this with all projects? Then you save the declaration above.

File: gradle/bintray.gradle [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    publish = true
+}
+
+
+private def validateTagAndVersionForPublishing() {
+    if (gitTag().isEmpty()) {
+        throw new IllegalStateException('Publishing is not allowed because current commit has no tag')
+    }
+
+
```
Comment (vanniktech) [inline gradle/bintray.gradle]: Spacing police

**Comment by artem-zinnatullin** [comment]
>Any reason why you're not uploading to mavenCentral?

Damn it, Niklas, totally forgot that we only use Bintray for private stuff. Uh. Lemme figure that out, likely will be pushing directly to Maven Central then and let JCenter cache that on their own

File: build.gradle (hunk 1 above)
Comment (artem-zinnatullin) [reply build.gradle]: It's not about Kotlin, JCenter was improperly proxying/caching some dependencies from Google's repo thus breaking builds, it should be solved now

File: build.gradle (hunk 2 above)
Comment (artem-zinnatullin) [reply build.gradle]: Will double check after I switch to Maven Central

**Comment by artem-zinnatullin** [comment]
🔒 (I know it doesn't work for public Lyft repos, but I'm kinda used to it)

File: gradle/publish-aar.gradle [hunk 4]
```diff
@@ -0,0 +1,51 @@
+apply plugin: 'maven-publish'
+
+
+publishing {
+    publications {
+        LibraryPublication(MavenPublication) {
+            groupId project.group
+            artifactId project.name
+            version project.version
+            artifact bundleRelease
```
Comment (artem-zinnatullin) [inline gradle/publish-aar.gradle]: Heads up, `bundleRelease` is changed to `bundleReleaseAar` in AGP 3.3.x

cc @vanniktech

Comment (vanniktech) [reply gradle/publish-aar.gradle]: Seen your tweet

**Pull request closed without merging**